crates/lovely-core/src/patch/pattern.rs

```diff
@@ -35,24 +35,97 @@ impl PatternPatch {
             return false;
         }
 
-        let wm = WildMatch::new(&self.pattern);
-        let mut matches = rope
-            .raw_lines()
-            .enumerate()
-            .map(|(i, line)| (i, line.to_string()))
-            .filter(|(_, line)| wm.matches(line.trim()))
-            .collect::<Vec<(_, _)>>();
+        let wm_lines = self
+            .pattern
+            .lines()
+            .map(|x| x.trim())
+            .map(WildMatch::new)
+            .collect::<Vec<_>>();
+        if wm_lines.is_empty() {
+            log::warn!("Pattern on target '{target}' has no lines");
+            return false;
+        }
+        let wm_lines_len = wm_lines.len();
+
+        let mut line_index = 0usize;
+        let rope_lines = rope.raw_lines().map(|x| x.to_string()).collect::<Vec<_>>();
+        let mut matches = Vec::new();
+        while let Option::Some(rope_window) = rope_lines.get(line_index..line_index + wm_lines_len)
+        {
+            if rope_window
+                .iter()
+                .zip(wm_lines.iter())
+                .all(|(source, target)| target.matches(source.trim()))
+            {
+                matches.push((
+                    line_index,
+                    if self.match_indent {
+                        String::from_utf8(
+                            rope_window[0]
+                                .bytes()
+                                .take_while(|x| *x == b' ' || *x == b'\t')
+                                .collect::<Vec<_>>(),
+                        )
+                        .unwrap()
+                    } else {
+                        String::new()
+                    },
```

Comment:
Replace this if statement with a proper var assign.

Reply:
Not entirely sure what you meant, but I modified the code to be more understandable.

Reply:
Great changes and my bad, I shouldn't do reviews when I'm tired.